server/src/app/assign-grades.js

```diff
@@ -312,26 +312,26 @@ export const scores = (req, res, agPayload, task) => {
       };
 
       request(options, function (err, response, body) {
-        let json = JSON.parse(body);
+        let json = body ? JSON.parse(body) : '';
 
         if (err) {
           //console.log('AGS Send Score Error - request failed: ' + err.message);
+        } else if (response.statusCode === 204) {
+          json = JSON.parse('{"status": "204", "message": "Score ignored as timestamp is earlier than the already graded date."}');
```

Comment:
This message isn't 100% accurate. https://www.imsglobal.org/spec/lti-ags/v2p0#score-service-media-type-and-schema indicates 204 is a valid response at any time, even if the score is accepted.

How about this?
```suggestion
        } else if (response.statusCode === 204) {
          json = {"status": 204, "message": "No content"};
```

Reply:
Hello @varju ,
Done in the latest commit.